schema/annotation-source.yaml

```diff
@@ -73,12 +73,12 @@ $defs:
         const: Contribution
         default: Contribution
         description: MUST be "Contribution".
-      agent:
+      contributor:
         $ref: "#/$defs/Agent"
       date:
         type: string
         format: date
-      role:
+      contributor_role:
```

Comment:
I am fine with the use of `contributor_role` over `role` if we are going the contributor role direction, though I think there is an unresolved issue as to whether we should adopt more of an `action` style contribution than a `role` style contribution. In CIViC, records are `created` by `contributor/agent` in a `curator` role, but records are also `revised` by a `contributor/agent` in a `curator` role. It seems off to me that the role changes based on the action taken and would actually prefer to see `role` changed to `action` instead. @mbrush is it possible that we can have _both_ `role` and `action`? That would be a-ok in my book!

Reply:
I think we really need the activity type (prefer activity over action). We can add it later. It is computationally complex to infer the activity from the role. I'm ok with contributing roles but it is less useful for my immediate needs.

Reply:
To clarify, the `contributor_role` in this context is the _role realized by the agent in the act of making their contribution_ (e.g. a submitter role, or editor role, or approver role).  It is specific to their act of contributing to an artifact.  It is _NOT the role/position they hold in their organization_, e.g. a 'curator position'  (but for the record the VA model does have a  dedicated place to capture this role/position should it be important). 

As I've stated, the main reason I am pushing framing this as a role is to be able to utilize the [Contributor Role Ontology (CRO)](https://github.com/data2health/contributor-role-ontology) that other folks in my group have worked on, and the [Contributor Attribution Model (CAM)](https://contributor-attribution-model.readthedocs.io/en/latest/) that uses CRO terms to describe the way agents contribute to research artifacts. With my community coordination and re-use hat on, I would like to align our models here - and we can really re-use CRO or claim CAM alignment if we don't use roles as proposed.

That said, I  understand the reason for confusion here, and the VA model would support your way of doing things if we simply use the existing `Contribution.contribution type` attribute to hold this info.  For example, to describe a submitter contribution, we would say:

````
type: Contribution
contribution_type: submission
contributor: Agent001
date: 2021-07-15
````

If you want can use this pattern for now, and remain VA compliant.  And consider shifting to use of the `contributor_role` attribute in future iterations should we aspire to use CRO and align with the CAM model.

Reply:
I read up on CRO, and understand the vision of this with respect to CRediT, etc. And I am supportive of the notion of using `contributor_role` to describe the broad types of roles a person may play in the creation of knowledge. However, I agree with @larrybabb that it adds complexity (and potentially reduces precision) to infer the action taken from the role of the contributor, and think it is important–and more correct–to also have contributions capture the contribution activity directly. It sounds like `contribution_type` is acceptable, though–is it okay to express both?

Reply:
OK - sounds like we will use contribution_type instead of contribution_role for now.  As for keeping both, I don't see what this adds, as in my mind these fields capture the exact same information (its just that one frames the value as an activity, and the other as a role realized by the contribution activity).  @ahwagner if you foresee these attributes capturing different types of information, could you provide an example so I can consider if we might keep both?

Reply:
I am going to stick with my [original comment](https://github.com/ga4gh/va-spec/pull/86/files#r961156780) on this as the example, which had a 👍 from @larrybabb so I am assuming he sees it the same way. You had mentioned that the role does not capture the organizational role and I get that, but `[curator](https://ontobee.org/ontology/CRO?iri=http://purl.obolibrary.org/obo/CRO_0000027)` is definitely a CRO role and I think the act of accepting / revising / rejecting curated records is still curation, but the corresponding roles for those specific actions (e.g. modifier_role, acceptor_role) are not descendent terms of `curator` so it is unclear that one would know this person was modifying e.g. the _curation_ of knowledge instead of the _generation_ of knowledge. It would be great of `CRO` had the notion of an "expert curator" so we could use CRO to recognize contributions from a person who plays the role of an acknowledged domain expert who is adding / modifying / accepting data in the process of knowledge curation, with the specific actions captured by `contribution_type`. Because at the end of the day (for CIViC, at least), each contribution captured by the system requires review and discussion of the same data, and the role shouldn't be anything other than "curator" just because a person clicks the "accept" button (i.e. agreement) vs. revising / rejecting the initial content vs. providing the initial content.

Reply:
My final thoughts here:
 
To start, I am not sure how/why the notion of a **'curator' role/action** came up - as it doesn’t seem to be relevant to our data (we don’t capture this as a contributor role/action in metakb or cvc data AFAIK).  And it is a challenging example on which to base our discussions because it is such a broad and variably used term - and one that is commonly used to describe an organizational job/position title, as well as a type of action that a person may perform (even if they are not officially a 'curator' in their organization). This is why I don't really like using this term to describe an agent's contribution.  Re: its representation in CRO - I do think some work is needed here to address the more specific types of contribution activities/roles that it subsumes. but that is a separate and not pertinent matter.
 
So if it pleases the court, lets completely set the conflated notion of 'curation/curator' aside as not relevant here.  For our purposes, we have already enumerated a defined set of ways that agents can contribute to cvc and metakb statements and other information entities.
 
For the cvc data, the key ones seem to be:
- framed as roles: `submitter`, `evaluator`, `approver`, and `transformer`.
- framed as actions: `submission`, `evaluation`, `approval`, `transformation`
 
For the CIViC data, the key ones seem to be (although I may be missing a couple):
- framed as roles: `submitter`, `reviewer`, `modifier`, `acceptor`
- framed as actions: `submission`, `review`, `modification`, `accepted`
 
We have a VA 'Contribution' pattern that could allow us to frame these _as roles realized in a contribution activity_, e.g.:
````
type: Contribution
contributor: Agent001
contributor_role: approver
date: 2021-07-15
 ````

Or we could chose a VA 'Contribution' pattern that lets us frame these as _a more specific type of contribution activity_, e.g.:

````
type: Contribution
contribution_type: approval
contributor: Agent001
date: 2021-07-15
 ````

While the semantics of these representations are subtly different, they get the same information across.  One pattern lets us use terms from an existing ontology (CRO) to carry this information, and align with the modeling pattern/semantics of the CAM model. The other pattern is still valid in VA - and seems to be more intuitive to metakb and cvc users. 
 
As I said, I am happy to go with either pattern at this point, and revisit when we have additional use cases. What I _don’t think we need_ in any given implementation model is to support both patterns - unless you can provide an example of when this might be needed or useful @ahwagner?

Finally, I'll note that if we did want to represent the organizational position/status that the agent held when they made a particular contribution, the VA  model provides a pattern for this as well - but this doesn't seem to be needed here so I won't get into it.